libsolidity/parsing/Token.cpp

```diff
@@ -101,17 +108,21 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-unsigned Token::extractM(string const& _literal)
+unsigned Token::extractUnsigned(string::const_iterator const& _begin, string::const_iterator const& _end)
 {
 	try
 	{
-		unsigned short m = stoi(_literal);
+		unsigned short m = stoi(string(_begin, _end));
```

Comment:
Ah sorry, did not see the string creation here. Could you use boost::lexical_cast<unsigned>()?